platform/ios/src/MGLFaux3DUserLocationAnnotationView.m

```diff
@@ -433,6 +464,74 @@ - (void)drawDot
 
         [self updateFaux3DEffect];
     }
+    
+}
+
+- (void)drawApproximate
+{
+    
+    if ( ! _approximateModeActivated)
+    {
+        self.layer.sublayers = nil;
+
+        _headingIndicatorLayer = nil;
+        _dotBorderLayer = nil;
+        _dotLayer = nil;
+        _accuracyRingLayer = nil;
+        _haloLayer = nil;
+        _puckDot = nil;
+        _puckArrow = nil;
+        
+        _approximateModeActivated = YES;
+    }
+
+    // update accuracy ring (if zoom or horizontal accuracy have changed)
+    if (_approximateLayer && (_oldZoom != self.mapView.zoomLevel || _oldHorizontalAccuracy != self.userLocation.location.horizontalAccuracy))
+    {
+        CGFloat borderWidth = 2;
+        if (self.mapView.zoomLevel < 7) {
```

Comment:
can we move this zoom level value into a const in case we have to change in the future
maybe like `const int maxApproximateZoomLevel = 7` <-- or w.e the correct obj-c syntax is